File: backend/app/schemas/pet_friendly.py
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID
from enum import Enum


class PetFriendlyCategory(str, Enum):
    cafeteria = "cafeteria"
    bar_restaurante = "bar_restaurante"
    hotel = "hotel"
    experiencia = "experiencia"


class PetFriendlyPlaceCreate(BaseModel):
    nombre: str
    categoria: PetFriendlyCategory
    lat: float
    lng: float
    descripcion: str | None = None
    foto_url: str | None = None


class PetFriendlyPlace(BaseModel):
    id: UUID
    nombre: str
    categoria: PetFriendlyCategory
    lat: float
    lng: float
    descripcion: str | None = None
    foto_url: str | None = None
    fuente: str
    verificado: bool
    created_by: UUID | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: backend-worker/src/schemas/petfriendly.ts
import { z } from "zod";
import { PaginatedResponse } from "./common.js";

export const categoriaEnum = z.enum([
  "cafeteria",
  "bar_restaurante",
  "hotel",
  "experiencia",
]);

export const CreatePetFriendlyPlaceRequest = z.object({
  nombre: z.string(),
  categoria: categoriaEnum,
  direccion: z.string().optional(),
  lat: z.number(),
  lng: z.number(),
  descripcion: z.string().optional().nullable(),
  foto_url: z.string().url().optional().nullable(),
});

export const PetFriendlyPlace = z.object({
  id: z.string().uuid(),
  nombre: z.string(),
  categoria: categoriaEnum,
  direccion: z.string().nullable().optional(),
  lat: z.number(),
  lng: z.number(),
  descripcion: z.string().nullable().optional(),
  foto_url: z.string().url().nullable().optional(),
  fuente: z.string(),
  verificado: z.boolean(),
  created_by: z.string().uuid().nullable().optional(),
  created_at: z.string(),
  updated_at: z.string().optional(),
});

export const PaginatedPetFriendlyPlaces = PaginatedResponse(PetFriendlyPlace);

export type CreatePetFriendlyPlaceRequest = z.infer<typeof CreatePetFriendlyPlaceRequest>;
export type PetFriendlyPlace = z.infer<typeof PetFriendlyPlace>;
